src/Client.cpp

```diff
@@ -591,6 +591,21 @@ bool CClient::PutClient(const CMessage& Message) {
     switch (Message.GetType()) {
         case CMessage::Type::Away:
             if (!m_bAwayNotify) return false;
+            // Do not send away-message if all shared channels are detached.
```

Comment:
This is a good idea, but it doesn't belong here: the channel is attached or detached for all clients together. none of this code is specific to the client. So if several clients are connected, you're doing the same check several times.

The check should happen in IRCSock.cpp instead